skaha/src/main/java/org/opencadc/skaha/session/PostAction.java

```diff
@@ -504,36 +506,35 @@ private String validateImage(String imageID, String type) throws Exception {
             }
         }
 
-        return validatedType;
+        return SessionType.fromApplicationStringType(validatedType);
     }
 
-    public void checkExistingSessions(String userid, String type) throws Exception {
+    public void checkExistingSessions(String userid, SessionType type) throws Exception {
         // multiple
-        if (SESSION_TYPE_HEADLESS.equals(type)) {
-            return;
-        }
-        List<Session> sessions = super.getAllSessions(userid);
-        int count = 0;
-        for (Session session : sessions) {
-            log.debug("checking session: " + session);
-            if (!SESSION_TYPE_HEADLESS.equalsIgnoreCase(session.getType())
-                    && !TYPE_DESKTOP_APP.equals(session.getType())) {
-                final String status = session.getStatus();
-                if (!(status.equalsIgnoreCase(Session.STATUS_TERMINATING)
-                        || status.equalsIgnoreCase(Session.STATUS_SUCCEEDED))) {
-                    count++;
+        if (!type.isHeadless()) {
+            List<Session> sessions = super.getAllSessions(userid);
+            int count = 0;
+            for (Session session : sessions) {
+                log.debug("checking session: " + session);
+                if (!SESSION_TYPE_HEADLESS.equalsIgnoreCase(session.getType())
+                        && !TYPE_DESKTOP_APP.equals(session.getType())) {
+                    final String status = session.getStatus();
+                    if (!(status.equalsIgnoreCase(Session.STATUS_TERMINATING)
+                            || status.equalsIgnoreCase(Session.STATUS_SUCCEEDED))) {
+                        count++;
+                    }
                 }
             }
-        }
-        log.debug("active interactive sessions: " + count);
-        if (count >= maxUserSessions) {
-            throw new IllegalArgumentException("User " + posixPrincipal.username + " has reached the maximum of "
-                    + maxUserSessions + " active sessions.");
+            log.debug("active interactive sessions: " + count);
+            if (count >= maxUserSessions) {
+                throw new IllegalArgumentException("User " + posixPrincipal.username + " has reached the maximum of "
```

Comment:
The method already receives `userid` as a parameter, so I am confused to why we are using `posixPrincipal.username` here?